### DISCUSSION TITLE
Ability to set project via env var

### DISCUSSION THREAD

**Discussion by DimmKirr** [ankitpokhrel/jira-cli#665] [Q&A]
Hey all,

I'm looking for a feature where I can set project key via env var. Is there something like this available?
I'd like to add `export JIRA_PROJECT_KEY=TST` to my .envrc so this repo is always connected to a `TST` project.

I know Viper supports autoenvs, but it doesn't seem there has been a replacer added, so dots should be dots in order for this to work, but `jira_project.key=TST` is not a valid bash/zsh thing.

I'm happy to send a PR for a replacer, but I'm sure there is a better way of doing so. Thanks!

**Comment by ankitpokhrel** [comment]
Hi @AutomationD, you can use `XDG_CONFIG_HOME` env or `--config/-c` option to specify which config to use. Does that works for your usecase?

**Reply by DimmKirr** [reply]
Hey! Not really, b/c I just have one config and would like to just switch projects.
Does anyone use the viper autoenv feature at all?  with `jira_` prefix? (lowerase)
If not, the solution is quite simple:
1. Change autoenv prefix `jira` to `JIRA`
2. add `SetEnvKeyReplacer(strings.NewReplacer(".", "_"))` for single underscore or `SetEnvKeyReplacer(strings.NewReplacer(".", "__"))` (for double underscore)

**Reply by DimmKirr** [reply]
This is what I ended up with: 
1. Create a jira config directory a repo: `.jira-config/`
2. Add it to global gitignore.
3. Add .envrc with `export XDG_CONFIG_HOME="$PWD/.jira-config/"`

But adding a proper string replacement would make more sense, IMO, since dots are not really useable.

**Reply by rodcloutier** [reply]
go-jira support hierarchy config reading. Maybe this something that could be supported as well. 
This is really convenient when working on several projects and basically allow to to "auto-switch" based on the sharable information in the directories. 

See https://github.com/go-jira/jira#configuration

**Reply by rodcloutier** [reply]
Hitting this issue again. 
@AutomationD solution of using the `SetEnvKeyReplacer` is quite simple and would enable user to use Env var for keys that are nested. 

@ankitpokhrel Would you mind if I create a PR for that? 

(Always funny when you circle back to your own comment...)
